server/src/main/java/org/opensearch/indices/IndicesRequestCache.java

```diff
@@ -437,8 +449,23 @@ public int hashCode() {
      * If Staleness threshold is 0, we do not keep track of stale keys in the cache
      * */
     class IndicesRequestCacheCleanupManager implements Closeable {
+        class StalenessProperty {
+            private int count;
+            private boolean isAccounted;
```

Comment:
this is not needed, we can rely on the entry being present or not in the staleKey map to decrement the stale counter. If absent, decrement it otherwise just reduce the value in the map